### PR TITLE
build(deps) upgrade angular

### PR DESCRIPTION
**Pull request by MatthijsSmets**
Upgrade to angular 17

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,33 +46,32 @@
     "font-awesome": "4.7.0",
     "jquery": "^3.6.0",
     "jqwidgets-ng": "^14.1.17",
-    "ngx-cookie-service": "^13.1.2",
-    "ngx-text-diff": "^0.6.0",
```
Comment (Matthbo) [inline package.json:49]: Did you forget to add ngx-text-diff back?
And if so, test it so it works properly as it did before but with angular 16

File: package.json [hunk 2]
```diff
@@ -41,53 +42,54 @@
...
-    "@angular-eslint/schematics": "14.4.0",
-    "@angular-eslint/template-parser": "14.4.0",
-    "@angular/cli": "14.2.10",
-    "@angular/compiler-cli": "14.2.12",
+    "@angular-devkit/build-angular": "^17.0.9",
+    "@angular-eslint/builder": "16.3.1",
+    "@angular-eslint/eslint-plugin": "16.3.1",
+    "@angular-eslint/eslint-plugin-template": "16.3.1",
+    "@angular-eslint/schematics": "16.3.1",
+    "@angular-eslint/template-parser": "16.3.1",
```
Comment (Matthbo) [inline package.json:65]: Shouldn't these match angular's major version?

File: src/app/compare/compare-data.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { NodeLinkStrategy } from '../shared/enums/compare-method';
+
+export class CompareData {
+  id!: string;
+  originalReport: any;
+  runResultReport: any;
```
Comment (Matthbo) [inline src/app/compare/compare-data.ts:6]: At some point we should really try to remove as many `any` types as possible as to have accurate and consistent types throughout the whole application

File: src/app/compare/compare.component.ts [hunk 4]
```diff
@@ -54,8 +47,8 @@ export class CompareComponent implements AfterViewInit {
   }
 
   saveDiff(leftSide: string, rightSide: string) {
-    this.diffComponent.left = leftSide;
-    this.diffComponent.right = rightSide;
-    this.diffComponent.renderDiffs();
+    // this.diffComponent.left = leftSide;
+    // this.diffComponent.right = rightSide;
+    // this.diffComponent.renderDiffs();
   }
```
Comment (Matthbo) [inline src/app/compare/compare.component.ts:53]: Is this function even needed if everything is commented out or are you gonna use it again at a later point?

File: src/app/text-compare/text-compare.component.css [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    border: 1px solid #ccc;
+    background-color: #ecebeb;
+}
+
+.input-title label {
+    margin-top: 6px;
+}
+
+.diff-editor {
+    height: 40vh !important;
```
Comment (Matthbo) [inline src/app/text-compare/text-compare.component.css:25]: Is it smart to use `vh` here and not a more precise measure or `%`?

File: .eslintrc.json [hunk 6]
```diff
@@ -8,7 +8,8 @@
   },
   "root": true,
   "ignorePatterns": [
-    "projects/**/*"
+    "projects/**/*",
+    "main.ts"
```
Comment (Matthbo) [inline .eslintrc.json:12]: Why exclude main.ts?

**Review by Matthbo — APPROVED** [review]
I have a few nit-picking questions which you can look into or explain, but I'll still approve

File: src/app/text-compare/text-compare.component.css (hunk 5 above)
Comment (MatthijsSmets) [reply src/app/text-compare/text-compare.component.css:25]: Yeah the idea is for the diff editor to take 40% of the height available in the visible window for the user

File: src/app/compare/compare-data.ts (hunk 3 above)
Comment (MatthijsSmets) [reply src/app/compare/compare-data.ts:6]: Yeah i agree, maybe the next big pr

File: src/app/compare/compare.component.ts (hunk 4 above)
Comment (MatthijsSmets) [reply src/app/compare/compare.component.ts:53]: I'm still figuring out what its use was, its not covered by the tests so its a little harder to find out

File: package.json (hunk 2 above)
Comment (MatthijsSmets) [reply package.json]: Yes you're right, fixed it.

File: .eslintrc.json (hunk 6 above)
Comment (MatthijsSmets) [reply .eslintrc.json:12]: fixed

**Pull request merged**